ctest-next/src/generation/generator.rs

```diff
@@ -0,0 +1,20 @@
+use std::{error::Error, path::Path};
+
+use crate::expand;
+
+pub struct TestGenerator;
```

Comment:
Make this `#[non_exhaustive] struct TestGenerator {}` so building this has to go through the constructor